RoboJackets/robocup-software
# Defense to waller refactor

## Description
This PR changes the behavior of any robot assigned as a defensive robot to act as a waller between the ball and the goal. The wall is not cooperative between the robots at the time of this PR, but multiple defenders will form a clump as seen in the gif later in this PR.

## Associated / Resolved Issue
https://app.clickup.com/t/86773u1ke
https://app.clickup.com/t/389zmma

## Steps to Test
1. Run Simulator
2. See any robot set as a defensive robot follow the ball outside of the perimeter, as shown by robots 1 and 2 in the gif below. You can move the ball with the mouse and watch the defensive robots follow it.

![ezgif-3-cdf34271c2](https://user-images.githubusercontent.com/68755425/214453696-ee10f5de-fb87-41ac-9c37-0eaf0ce49b31.gif)


## Key Files to Review
Defense.cpp
Defense.hpp
